common/appscale/common/monit_interface.py

```diff
@@ -248,14 +257,74 @@ def send_command(self, process_name, command):
       command: A string specifying the command to send.
     """
     process_url = '{}/{}'.format(self.LOCATION, process_name)
-    payload = urllib.urlencode({'action': command})
+    params = {'action': command}
+
+    headers = {}
+    if self._csrf_token is not None:
+      headers['Cookie'] = 'securitytoken={}'.format(self._csrf_token)
+      params['securitytoken'] = self._csrf_token
+
     try:
-      yield self._async_client.fetch(process_url, method='POST', body=payload)
+      yield self._async_client.fetch(
+        process_url, method='POST', body=urllib.urlencode(params),
+        headers=headers)
     except HTTPError as error:
-      if error.code == 404:
+      if error.code == httplib.FORBIDDEN:
+        # Retrieve CSRF token (introduced in Monit 5.20).
+        response = yield self._async_client.fetch(process_url)
+        self._csrf_token = re.search('securitytoken=([a-zA-Z0-9]+);',
```

Comment:
Not sure if we'll hit it with monit, but the Cookie header syntax doesn't require a semicolon, so its possible (probably unlikely) to see:
```
Cookie: securitytoken=12345
```
too bad the client doesn't have support for cookie handling/parsing. oh well.

Reply:
Good point. Using Python's cookie library is more robust.